### ISSUE TITLE
release-5.5.x/dev-image job is breaking

### ISSUE THREAD

**Issue by muntac** [concourse/ci#293]
https://ci.concourse-ci.org/teams/main/pipelines/release-5.5.x/jobs/dev-image/builds/58

```
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to build LLB: failed to compute cache key: "/runc/runc.amd64" not found: not found
FATA[0020] failed to build: build: exit status 1        
FATA[0020] failed to run task: exit status 1         
```

**Comment by jomsie** [comment]
So since the new containerd support was added in 5.8.x, any release pipelines (which still read from concourse/ci's master branch) will break due to certain inputs (containerd, runc, cni, etc) not being configured or available on those release branches. (Definitely referring to any pre 5.8.x release pipelines when containerd wasn't a thing in concourse yet). 

The first issue that we have seen is to do with building the concourse/dev docker image. The dockerfile itself is looking to copy some runc related files, but since the release pipeline isn't configured with the new containerd resources, they won't be there.

We had a couple of ideas, such as just bundling those binaries into the release-5.5.x release, or having some new variables being passed into the set pipeline for a release pipeline, but when talking to Jamie about it, we reconsidered. Bundling those binaries in would cause a lot of unnecessary bloat, for instance. 

Our current working branch: https://github.com/concourse/ci/tree/fix-release-pipeline

Our current way of thinking would be to have a few legacy files representing the old way of building concourse. No containerd, no runc, no cni, etc. It seems like a pretty quick way to unblock the release pipelines, since they'll probably be used shortly.

So in the case of a pre 5.8.x release pipeline, one would do a `fly sp -p release-legacy.yml ...` instead of `fly sp -p release.yml ...`, and that release-legacy.yml would have different task definitions used throughout, depending if the containerd changes influenced the pipeline.

This change would only affect older release pipelines, and going forward it wouldn't change the way new release pipelines would be created.

We're still open to figuring out another way, if it's cleaner, but to get things unblocked sooner, we've gone this route.

**Comment by muntac** [comment]
We were solving this in https://github.com/concourse/ci/pull/295 but decided the approach in https://github.com/concourse/ci/pull/294 has less duplication, and are going with that. It has some changes to be added and will be good to go after that.

**Comment by pivotal-jamie-klassen** [comment]
fixed in #294
